src/shrinkray/reducer.py

```diff
@@ -117,14 +136,90 @@ class RestartPass(Exception):
     pass
 
 
+@define
+class PassStats:
+    """Statistics for a single reduction pass."""
+
+    pass_name: str
+    bytes_deleted: int = 0
+    non_size_reductions: int = 0
+    call_count: int = 0
+    test_evaluations: int = 0
+    successful_reductions: int = 0
+    order: int = 0  # Track insertion order for stable display
+
+    @property
+    def success_rate(self) -> float:
+        """Percentage of test evaluations that led to a reduction."""
+        if self.test_evaluations == 0:
+            return 0.0
+        return (self.successful_reductions / self.test_evaluations) * 100.0
+
+
+@define
+class PassStatsTracker:
+    """Tracks statistics for all reduction passes."""
+
+    _stats: dict[str, PassStats] = attrs.Factory(dict)
+    _next_order: int = 0
+
+    def get_or_create(self, pass_name: str) -> PassStats:
+        if pass_name not in self._stats:
+            self._stats[pass_name] = PassStats(
+                pass_name=pass_name, order=self._next_order
+            )
+            self._next_order += 1
```

Comment:
Why are you tracking this given that dictionary values are in insert order anyway?